### PR TITLE
[FEAT] Add initial setup for API

### LINKED ISSUE #52: [FEAT] Initial Setup for API

### PR DESCRIPTION
**Pull request by Jaysce**
- Add mongoose Schema model for User and Session
- Add /sessions and /users routes for API

### CODE REVIEW

File: backend/src/index.js [hunk 1]
```diff
@@ -1,48 +1,53 @@
-import express from 'express';
```
Comment (David-Xia0) [inline backend/src/index.js:1]: Hey, I think you are running the wrong Linter.  
https://github.com/we-are-number-1/yumble/blob/main/CONTRIBUTING.md. We have decided to use ES Lint :).

File: backend/models/Session.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import mongoose from "mongoose";
```
Comment (David-Xia0) [inline backend/models/Session.js:1]: Also, regarding package structure, could you potentially do mongo/models/*.js.
Just sot here is no confusion with the domain models. 🙃

**Comment by BlinkCreator** [comment]
Hey Sasindu, Thanks for adding the setup for the api, just a reminder to assign yourself to the issue and to read the CONTRIBUTING.md, nice work on your first PR.

File: backend/src/index.js (hunk 1 above)
Comment (BlinkCreator) [reply backend/src/index.js:1]: Hey, I noticed the same thing, do you think it would be possible to create pre-commit hooks to test the es linter?

Comment (David-Xia0) [reply backend/src/index.js:1]: hmm, will consult with Alex, Good idea !

Comment (Jaysce) [reply backend/src/index.js]: ES Lint complains about express' Router function as it is using a capital letter. Should we just ignore these warnings or is there a way to suppress certain warnings?
Screen Shot 2021-03-14 at 2 43 00 pm
Screen Shot 2021-03-14 at 2 43 12 pm

File: backend/models/Session.js (hunk 2 above)
Comment (Jaysce) [reply backend/models/Session.js]: Fixed.

File: backend/src/index.js (hunk 1 above)
Comment (David-Xia0) [reply backend/src/index.js]: Leave it for now, I will ask Alex, he set up the linter. There seems to be a lot of other problems with it as well

**Review by David-Xia0 — APPROVED** [review]
I will consult Alex about other comments.

File: backend/src/index.js (hunk 1 above)
Comment (Alxjlv) [reply backend/src/index.js]: David has added an ESLint rule to ignore that specific issue

**Review by David-Xia0 — APPROVED** [review]

**Review by Alxjlv — DISMISSED** [review]

File: backend/src/mongo/models/Preference.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    required: true,
+  },
+  cuisines: {
+    type: [String],
+    default: [],
+  },
+  price: {
+    type: Number,
+    required: true,
+  },
```
Comment (BlinkCreator) [inline backend/src/mongo/models/Preference.js:23]: I like your eagerness to get into the code, but can we just make sure your api contract is finished and approved before pushing this?

image

Comment (BlinkCreator) [reply backend/src/mongo/models/Preference.js:23]: As far as I can tell it looks correct, but just create a sample payload using mock data so everyone has access to the same test case

File: backend/src/mongo/models/Restaurant.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
+import mongoose from 'mongoose';
+
+export const restaurantSchema = mongoose.Schema({
+  name: {
+    type: String,
+    required: true,
+  },
+  numberOfVotes: {
+    type: Number,
+    default: 0,
```
Comment (BlinkCreator) [inline backend/src/mongo/models/Restaurant.js:10]: I think we're putting numberOfVotes into the results schema. @cecilia-yin , @lilichenyuxin . What naming convention did you guys have in mind for these fields?

File: backend/src/mongo/models/Session.js [hunk 5]
```diff
@@ -0,0 +1,10 @@
+import mongoose from 'mongoose';
+
+const sessionSchema = mongoose.Schema({
+  isFinished: {
+    type: Boolean,
+    default: false,
```
Comment (BlinkCreator) [inline backend/src/mongo/models/Session.js:6]: Again I don't think this is complete

**Review by BlinkCreator — CHANGES_REQUESTED** [review]
@Jaysce  To get pr merged:
1. Finish api contract with sample test
2. Remove schemas not pertaining to your api so other teams can fill it once contracts have been established

**Comment by BlinkCreator** [comment]
image
Just a reminder that according to how to contribute all pr's and issues must be assigned to oneself

**Review by lilichenyuxin — DISMISSED** [review]

File: backend/src/mongo/models/Restaurant.js (hunk 4 above)
Comment (cecilia-yin) [reply backend/src/mongo/models/Restaurant.js]: Those names look fine by us.

Comment (lilichenyuxin) [reply backend/src/mongo/models/Restaurant.js:10]: Yep, We are agree with this, and we are using restaurant and numberOfVotes.

File: backend/src/mongo/models/Preference.js (hunk 3 above)
Comment (Jaysce) [reply backend/src/mongo/models/Preference.js:23]: Done.

File: backend/src/mongo/models/Restaurant.js (hunk 4 above)
Comment (Jaysce) [reply backend/src/mongo/models/Restaurant.js]: I have removed the Restaurant model, so one of you guys can implement it.

File: backend/src/mongo/models/Session.js (hunk 5 above)
Comment (Jaysce) [reply backend/src/mongo/models/Session.js]: I have removed the Session model, so one of you guys can implement it.

File: backend/src/mongo/models/Preference.js [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+  distance: {
+    type: Number,
+    required: true,
+  },
+  cuisines: {
+    type: [String],
+    default: [],
+  },
+  price: {
+    type: Number,
```
Comment (BlinkCreator) [inline backend/src/mongo/models/Preference.js:21]: This has been updated into an array

Comment (Jaysce) [reply backend/src/mongo/models/Preference.js]: Fixed in new commit

**Review by blakehattingh — APPROVED** [review]
Everything looks good, nice work with the set up.

**Review by cecilia-yin — APPROVED** [review]
LGTM

**Review by BlinkCreator — APPROVED** [review]
Looks good, life's good

**Comment by BlinkCreator** [comment]
Merged getPreferences to backend

**Pull request merged**